api_tests/metaschemas/views/test_metaschemas_detail.py

```diff
@@ -1,52 +1,54 @@
-from nose.tools import *  # flake8: noqa
-
-from website.project.metadata.schemas import LATEST_SCHEMA_VERSION
-from website.project.model import Q
-from osf.models import MetaSchema
+import pytest
 
 from api.base.settings.defaults import API_BASE
-
-from tests.base import ApiTestCase
+from osf.models import MetaSchema
 from osf_tests.factories import (
-    AuthUserFactory
+    AuthUserFactory,
 )
-class TestMetaSchemaDetail(ApiTestCase):
-    def setUp(self):
-        super(TestMetaSchemaDetail, self).setUp()
-        self.user = AuthUserFactory()
-        self.schema = MetaSchema.find_one(Q('name', 'eq', 'Prereg Challenge') & Q('schema_version', 'eq', LATEST_SCHEMA_VERSION))
-        self.url = '/{}metaschemas/{}/'.format(API_BASE, self.schema._id)
-
-    def test_pass_authenticated_user_can_retrieve_schema(self):
-        res = self.app.get(self.url, auth=self.user.auth)
-        assert_equal(res.status_code, 200)
+from website.project.metadata.schemas import LATEST_SCHEMA_VERSION
+
+@pytest.mark.django_db
+class TestMetaSchemaDetail:
+
+    def test_metaschemas_detail_visibility(self, app):
+
+        user = AuthUserFactory()
+        schema = MetaSchema.objects.filter(name='Prereg Challenge', schema_version=LATEST_SCHEMA_VERSION).first()
+
+        def url_metaschemas_detail(schema_id):
```

Comment:
Defining this function inside of this test is a little unusual. Either put it outside the test scope or just do it inline since it is a one-liner